### PR TITLE
Edit documentation

### LINKED ISSUE #335: End-user documentation
The following documentation must be provided:
- installation and usage instructions
- troubleshooting (especially the workarounds for assembly version conflicts and missing assemblies, possible telemetry loss)
- supported env (OS and frameworks)
- list of provided instrumentation libraries 
- configuration
- manual instrumentation instructions
- changelog

It would be also good to improve the contributing docs.

### PR DESCRIPTION
**Pull request by theletterf**
Fixes #335

Various edits in existing docs.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -61,8 +60,9 @@ aspNetAppTargetFramework="net6.0" ./run-example.sh
 
 ### BindingRedirect
 
-Example client app that uses .NET Framework binding redirects to handle assembly version conflict
```
Comment (Kielek) [inline examples/README.md:64]: In this repo Sample application means application used in tests. Example application is app just to illustrate something, placed in different folder.

I hope you find a way to rephrase to differentiate the usage. Alternatively, we can try to change the code if you have better names :)

Comment (Kielek) [reply examples/README.md:64]: #484

File: docs/config.md [hunk 2]
```diff
@@ -134,32 +125,32 @@ public OpenTelemetry.Trace.TracerProviderBuilder ConfigureTracerProvider(OpenTel
...
+| `CORECLR_PROFILER_PATH_64` | Path to the 64-bit profiler. | `%InstallationLocation%/win-x64/OpenTelemetry.AutoInstrumentation.Native.dll` for Windows |
 
-Remarks:
+### Considerations on the CLR profiler
 
-1. .NET Framework uses `CLR_` prefix instead of `CORECLR_`.
-2. The bitness-specific variables take precedence.
-3. The `*_PROFILER_PATH_*` environment variable is not needed on Windows
-   if the `.dll` is registered.
+- .NET Framework uses the `CLR_` prefix instead of `CORECLR_`.
```
Comment (pjanotti) [inline docs/config.md:146]: The prefix is actually `COR_`, see https://docs.microsoft.com/en-us/dotnet/framework/unmanaged-api/profiling/setting-up-a-profiling-environment

Comment (pjanotti) [reply docs/config.md:146]: What about moving this comment about the prefix to some place above the table?

File: examples/README.md [hunk 3]
```diff
@@ -17,41 +17,40 @@ You can see the traces generated by the example in the Jaeger container.
...
 
-The script run can be customized using the following environment variables:
+You can customize the script using the following environment variables:
 
 | Environment variable | Description | Default |
 |-|-|-|
 | **aspNetAppTargetFramework** | Target framework for the `aspnet-server` service | `netcoreapp3.1` |
 | **configuration** | Build configuration for example apps, either `Release` or `Debug` | `Release` |
-| **enableProfiling** | Value of `CLR_ENABLE_PROFILING` and `CORECLR_ENABLE_PROFILING` for the example apps | `1` |
+| **enableProfiling** | Value of `CLR_ENABLE_PROFILING` and `CORECLR_ENABLE_PROFILING` for the sample apps | `1` |
```
Comment (pjanotti) [inline examples/README.md:33]:
```suggestion
| **enableProfiling** | Value of `COR_ENABLE_PROFILING` and `CORECLR_ENABLE_PROFILING` for the sample apps | `1` |
```

File: examples/README.md [hunk 4]
```diff
@@ -75,11 +75,11 @@ OTEL_DOTNET_AUTO_LEGACY_SOURCES="InstrumentedHttpCall.GetAsync" exampleApp=Bindi
 
 ### ConsoleApp
 
-The default example app used by `./run-example.sh`. It contains manual instrumentation using an
+The default sample app used by `./run-example.sh`. It contains manual instrumentation which use an
```
Comment (pjanotti) [inline examples/README.md:78]: @Kielek per comment in the other PR about the manual instrumentation example, the info about manual instrumentation on the ConsoleApp (default example app) should be removed.

File: examples/README.md [hunk 5]
```diff
@@ -75,11 +75,11 @@ OTEL_DOTNET_AUTO_LEGACY_SOURCES="InstrumentedHttpCall.GetAsync" exampleApp=Bindi
...
 [ActivitySource](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/README.md#activity-source)
 and [OpenTracing](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Shims.OpenTracing#readme).
-No configuration is needed to capture the manually created ActivitySource, as its name is
-`OpenTelemetry.AutoInstrumentation.ConsoleApp`, which is match for the default value for the environment variable
+You don't need to add configuration to capture the manually created ActivitySource, as its name is
```
Comment (pjanotti) [inline examples/README.md:81]: Same here.

**Review by pjanotti — COMMENTED** [review]
Thanks @theletterf great improvements to the readability.

File: docs/config.md (hunk 2 above)
Comment (theletterf) [reply docs/config.md:146]: Done!

File: docs/README.md [hunk 6]
```diff
@@ -1,30 +1,29 @@
-# OpenTelemetry .NET Auto-Instrumentation
+# OpenTelemetry .NET Automatic Instrumentation
```
[the remark this answers is not in the document]
Comment (theletterf) [reply docs/README.md:1]: @pjanotti I'd say it's OK to keep the names as they are while spelling them out in full in the documentation, unless there's a 1:1 equivalence rule between headings in docs and library names that I'm now aware of.

**Review by Kielek — APPROVED** [review]

File: docs/README.md [hunk 7]
```diff
@@ -1,30 +1,29 @@
...
 2. Adds [OpenTelemetry Instrumentation](https://opentelemetry.io/docs/concepts/instrumenting/)
    to key packages and APIs used by the application.
 
-In addition, you can use the OpenTelemetry .NET Auto-Instrumentation as a .NET Profiler
-to inject additional instrumentations at runtime, using a technique known as [monkey-patching](https://en.wikipedia.org/wiki/Monkey_patch).
-This lets you instrument specific packages or APIs that does not provide the necessary hooks
-to generate .NET instrumentation packages using `System.Diagnostics.DiagnosticSource`.
+In addition, you can enable the OpenTelemetry .NET Automatic Instrumentation as a .NET Profiler
+to inject additional instrumentations at runtime, using a technique known as [monkey-patching](https://en.wikipedia.org/wiki/Monkey_patch). When enabled, the OpenTelemetry .NET Automatic Instrumentation generates traces for libraries that don't
+already generate traces using the OpenTelemetry .NET SDK.
```
Comment (nrcventura) [inline docs/README.md:23]: Should we call out that this additional instrumentation must be a part of this project? In other words there isn't a way for them to dynamically add this type of instrumentation for arbitrary libraries.

File: docs/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (pjanotti) [reply docs/README.md:1]: I'm fine with that @theletterf I just want to ensure that all the approvers are also aware of that.

File: docs/README.md (hunk 7 above)
Comment (theletterf) [reply docs/README.md]: Clarified! Thanks!

**Comment by theletterf** [comment]
@pjanotti @Kielek Conflicts solved. LMK when this can be merged.

**Review by Kielek — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Review by pjanotti — APPROVED** [review]

**Review by nrcventura — APPROVED** [review]

File: docs/README.md [hunk 8]
```diff
@@ -1,30 +1,31 @@
...
 
 This project adds [OpenTelemetry instrumentation](https://opentelemetry.io/docs/concepts/instrumenting/#automatic-instrumentation)
 to .NET applications without having to modify their source code.
 
-OpenTelemetry .NET Auto-Instrumentation is built on top of [OpenTelemetry .NET](https://github.com/open-telemetry/opentelemetry-dotnet):
+OpenTelemetry .NET Automatic Instrumentation is built on top of [OpenTelemetry .NET](https://github.com/open-telemetry/opentelemetry-dotnet):
 
-- core components: [`1.2.0-rc5`](https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/core-1.2.0-rc5)
-- non-core components: [`1.0.0-rc9.2`](https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/1.0.0-rc9.2)
+- Core components: [`1.2.0-rc5`](https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/core-1.2.0-rc4)
```
Comment (rajkumar-rangaraj) [inline docs/README.md:10]: GitHub URL is referring to older version, this line and next line needs a fix.

Comment (theletterf) [reply docs/README.md]: Fixed

**Review by rajkumar-rangaraj — APPROVED** [review]

**Pull request merged**